src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1PlacementMatchingOutcomesV2ReportTest.kt

```diff
@@ -0,0 +1,299 @@
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.cas1.Cas1RequestForPlacementReportTest.RequestForPlacementReportRow
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.givens.`Given a User`
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.givens.`Given an Offender`
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.httpmocks.APDeliusContext_mockSuccessfulCaseDetailCall
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.httpmocks.GovUKBankHolidaysAPI_mockSuccessfullCallWithEmptyResponse
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApplicationRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApprovedPremisesApplicationEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApprovedPremisesApplicationJsonSchemaEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApprovedPremisesAssessmentJsonSchemaEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserQualification
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserRole
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.asCaseDetail
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.UUID
+
+class Cas1PlacementMatchingOutcomesV2ReportTest : InitialiseDatabasePerClassTestBase() {
+
+  @Autowired
+  lateinit var realApplicationRepository: ApplicationRepository
+
+  @Autowired
+  lateinit var cas1SimpleApiClient: Cas1SimpleApiClient
+
+  lateinit var applicationSchema: ApprovedPremisesApplicationJsonSchemaEntity
+  lateinit var assessmentSchema: ApprovedPremisesAssessmentJsonSchemaEntity
+
+  lateinit var assessor: UserEntity
+  lateinit var assessorJwt: String
+
+  val standardRFPNotAssessed = StandardRFPNotAllocated()
+
+  object Constants {
+    const val REPORT_MONTH = 1
+    const val REPORT_YEAR = 2020
+  }
+
+  @BeforeAll
+  fun setup() {
+    GovUKBankHolidaysAPI_mockSuccessfullCallWithEmptyResponse()
+
+    val assessorDetails = `Given a User`(
+      roles = listOf(UserRole.CAS1_ASSESSOR, UserRole.CAS1_MATCHER),
+      qualifications = UserQualification.entries,
+      staffUserDetailsConfigBlock = {
+        withUsername("ASSESSOR1")
+      },
+    )
+    assessor = assessorDetails.first
+    assessorJwt = assessorDetails.second
+
+    applicationSchema = approvedPremisesApplicationJsonSchemaEntityFactory.produceAndPersist { withDefaults() }
+    assessmentSchema = approvedPremisesAssessmentJsonSchemaEntityFactory.produceAndPersist { withDefaults() }
+
+    standardRFPNotAssessed.createRequestForPlacement()
+  }
+
+  @Test
+  fun `Get report is empty if no applications`() {
+    `Given a User`(roles = listOf(UserRole.CAS1_REPORT_VIEWER)) { _, jwt ->
+
+      webTestClient.get()
+        .uri(getReportUrl(year = 2019, month = 2, includePii = true))
```

Comment:
would the setup be clearer here if you also used constants, as you do with the "correct applications" example?

e.g. `.uri(getReportUrl(year = BLANK_YEAR, month = BLANK_MONTH, includePii = true))`

Reply:
Good idea, i'll change that

FYI - i should have left this PR as draft as SQL is still work in progress (apologies), but TBH i think you've reviewed the key changes as the additional SQL will just result in more test code